contracts/impls/apps/WitnetPriceFeedsUpgradable.sol

```diff
@@ -476,15 +479,18 @@ contract WitnetPriceFeedsUpgradable
             // solhint-disable-next-line avoid-low-level-calls
             (bool _success, bytes memory _reason) = solver.delegatecall(abi.encodeWithSelector(
                 IWitnetPriceSolver.validate.selector,
-                feedId
+                feedId,
+                deps
             ));
-            require(
-                _success,
-                string(abi.encodePacked(
+            if (!_success) {
+                assembly {
+                    _reason := add(_reason, 4)
+                }
+                revert(string(abi.encodePacked(
                     "WitnetPriceFeedUpgradable: solver validation failed: ",
-                    _reason
-                ))
-            );
+                    string(abi.decode(_reason,(string)))
```

Comment:
Just out of curiosity, why is this decode needed?

Reply:
Because of the way the result to low-level calls is formatted: after jumping over the first 4 bytes, a `bytes` type comes, in which first word contains the length of such `bytes` array. When a revert is produced within the low-level call, the `bytes` actually contains the string message, the one that we need to the revert message that's being composed at the WitnetPriceFeedsUpgradable level.